assets/src/higher-order-components/local-storage/with-default-dates-list-settings.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * External imports
+ */
+import { createHigherOrderComponent } from '@wordpress/compose';
+
+/**
+ * Internal imports
+ */
+import useLocalStorage from './use-local-storage';
+
+const withDefaultDatesListSettings = createHigherOrderComponent(
+	( EntityListWithFilterBar ) => ( { ...otherProps } ) => {
+		const [
+			defaultDatesListView,
+			setDefaultDatesListView,
+		] = useLocalStorage( 'defaultDatesListView' );
+		const [
+			defaultDatesListPerPage,
+			setDefaultDatesListPerPage,
+		] = useLocalStorage( 'defaultDatesListPerPage' );
+
+		return (
+			<EntityListWithFilterBar
+				{ ...otherProps }
+				defaultDatesListView={ defaultDatesListView || 'grid' }
+				setDefaultDatesListView={ setDefaultDatesListView }
+				defaultDatesListPerPage={ defaultDatesListPerPage || 6 }
+				setDefaultDatesListPerPage={ setDefaultDatesListPerPage }
+			/>
```

Comment:
Rather than adding additional props, I think this should override the props passed through to whatever implements this.  So for example, you'd end up with something like:

```jsx
( EntityListWithFilterBar ) => ( {
	datesListView,
	datesListPerPage,
	...otherProps
} ) => {
	const [
		defaultDatesListView,
		setDefaultDatesListView,
	] = useLocalStorage( 'defaultDatesListView' );
	const [
		defaultDatesListPerPage,
		setDefaultDatesListPerPage,
	] = useLocalStorage( 'defaultDatesListPerPage' );

	return (
		<EntityListWithFilterBar
			{ ...otherProps }
			datesListView={ datesListView || defaultDatesListView }
			setDefaultDatesListView={ setDefaultDatesListView }
			datesListPerPage={ datesListPerPage || defaultDatesListPerPage }
			setDefaultDatesListPerPage={ setDefaultDatesListPerPage }
		/>
	);
};
```

But also, if you implement the persistence plugin for the filter state as I referenced in a previous comment I think this hoc would become unnecessary.

_note: comments here apply to the similar work done for tickets list_